cnf-tests/testsuites/e2esuite/dpdk/numa_node_sriov.go

```diff
@@ -142,6 +142,58 @@ var _ = Describe("[sriov] NUMA node alignment", Ordered, func() {
 		By("Create server Pod and run E2E ICMP validation")
 		validateE2EICMPTraffic(pod, fmt.Sprintf(`[{"name": "test-numa-1-exclude-topology-true-network","ips":["192.0.2.250/24"]}]`))
 	})
+
+	It("Utilize all available VFs then create a pod with guaranteed CPU and excludeTopology set to True", func() {
+		podDefinition := pods.DefinePod(sriovnamespaces.Test)
+		podDefinition = pods.RedefineWithGuaranteedQoS(podDefinition, "2", "500Mi")
+		podDefinition = pods.RedefinePodWithNetwork(podDefinition, "test-numa-0-exclude-topology-true-network")
+
+		By("Utilize all available VFs")
+		var expectedRunningPods []*corev1.Pod
+		for i := 0; i < 4; i++ {
```

Comment:
can you just create a pod requesting 5 devices?

Reply:
also please use the numVF variable :)

Reply:
I have to implement a step where all VFs are in use, so at least I can do the following:

```
create a pod with 4 VFs
assert it's running

create a pod with 1 VF (or more)
assert it raises an error
```

If I create a variable `numVFs` , I still need a loop to create the network annotation:

```go
                ...
                podDefinition := pods.DefinePod(sriovnamespaces.Test)
		podDefinition = pods.RedefineWithGuaranteedQoS(podDefinition, "2", "500Mi")

		numVFs := 4
		By("Utilize all available VFs")
		expectedRunningPod := podDefinition.DeepCopy()
		useAllVFsNetworkSpec := []string{}
		for i := 0; i < numVFs; i++ {
			useAllVFsNetworkSpec = append(useAllVFsNetworkSpec, "test-numa-0-exclude-topology-true-network")
		}
		expectedRunningPod = pods.RedefinePodWithNetwork(expectedRunningPod, strings.Join(","))

		expectedRunningPod, err := client.Client.Pods(sriovnamespaces.Test).
			Create(context.Background(), podDefinition, metav1.CreateOptions{})
		Expect(err).ToNot(HaveOccurred())

		Eventually(func(g Gomega) {
			actualPod, err := client.Client.Pods(sriovnamespaces.Test).Get(context.Background(), expectedRunningPod.Name, metav1.GetOptions{})
			g.Expect(err).ToNot(HaveOccurred())
			g.Expect(actualPod.Status.Phase).To(Equal(corev1.PodRunning))
			g.Expect(actualPod.Status.QOSClass).To(Equal(corev1.PodQOSGuaranteed))
		}, 30*time.Second, 1*time.Second).Should(Succeed())

		By("Create a failing pod that uses a VF")
		failingPod := podDefinition.DeepCopy()
		failingPod = pods.RedefinePodWithNetwork(failingPod, "test-numa-0-exclude-topology-true-network")
		failingPod, err = client.Client.Pods(sriovnamespaces.Test).
			Create(context.Background(), failingPod, metav1.CreateOptions{})
		Expect(err).ToNot(HaveOccurred())
                ...
```

Is that what you had in mind?

Reply:
yep I think is nicer then having 5 pods.
we have the same in another sriov test https://github.com/openshift/sriov-network-operator/blob/master/test/conformance/tests/test_sriov_operator.go#L837